pkg/binding/bindingpolicy.go

```diff
@@ -113,16 +129,33 @@ func (c *Controller) requeueForBindingPolicyChanges() error {
 	return c.requeueWorkloadObjects()
 }
 
-func (c *Controller) getBindingPolicyByName(name string) (runtime.Object, error) {
-	lister := c.listers["control.kubestellar.io/v1alpha1/BindingPolicy"]
-	if lister == nil {
-		return nil, fmt.Errorf("could not get lister for placememt")
+func (c *Controller) requeueSelectedWorkloadObjects(bindingPolicyName string) error {
+	if !c.bindingPolicyResolver.ResolutionExists(bindingPolicyName) {
+		return nil
 	}
-	got, err := lister.Get(name)
+
+	// requeue all objects that are selected by the bindingpolicy (are in its resolution)
+	objectKeys, err := c.bindingPolicyResolver.GetObjectKeys(bindingPolicyName)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed to get object keys from bindingpolicy resolver for bindingpolicy %s: %w",
+			bindingPolicyName, err)
+	}
+
+	objs := make([]runtime.Object, 0, len(objectKeys))
+	for _, key := range objectKeys {
+		obj, err := c.getObjectFromKey(*key)
+		if err != nil {
+			return fmt.Errorf("failed to get object from key: %w", err)
+		}
+
+		objs = append(objs, obj)
+	}
+
+	for _, obj := range objs {
+		c.enqueueObject(obj, true)
```

Comment:
Why two separate loops over the same set of objects? Why not fuse the two loops?

Reply:
Because in that case, it is probable that a failing object will keep failing for a number of retries. All objects iterated over before it would keep getting queued up.

Therefore I chose to enqueue all workload objects or none.